### PR TITLE
Room display name is now computed by the Matrix SDK

### PR DESCRIPTION
**Pull request by bmarty**
This PR also add test on the Room name computation

CC @manuroe

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    RoomMember currentUser = activeMembers.get(0);
+
+                    for (RoomMember roomMember : activeMembers) {
+                        if (roomMember.getUserId().equals(mRoom.getDataHandler().getUserId())) {
+                            currentUser = roomMember;
+                            break;
+                        }
+                    }
+
+                    if (TextUtils.equals(currentUser.membership, RoomMember.MEMBERSHIP_INVITE)) {
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java]: This check is already done by `mRoom.isInvited()`
You should remove it

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                    }
+                });
+            }
+
+            nbOfOtherMembers = othersActiveMembers.size();
+
+            String displayName;
+
+            if (mRoom.isInvited()) {
+                if (othersActiveMembers.size() == 1) {
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java]: You should check here "if (othersActiveMembers.size() > 0)"

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                return roomState.getCanonicalAlias();
+            }
+
+            // Temporary patch: use the first alias if available
+            if (roomState.aliases != null && !roomState.aliases.isEmpty()) {
+                return roomState.aliases.get(0);
+            }
+
+            List<RoomMember> othersActiveMembers = new ArrayList<>();
+            List<RoomMember> activeMembers = new ArrayList<>();
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java]: `activeMembers` array is not really useful now
You should remove it, and just look for the currentUserMember (RoomMember)

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                }
+            } else {
+                Collection<RoomMember> members = roomState.getDisplayableLoadedMembers();
+
+                for (RoomMember member : members) {
+                    if (!TextUtils.equals(member.membership, RoomMember.MEMBERSHIP_LEAVE)) {
+                        if (!TextUtils.equals(member.getUserId(), mRoom.getDataHandler().getUserId())) {
+                            othersActiveMembers.add(member);
+                        }
+                        activeMembers.add(member);
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java]: Remove this line with
else {currentUserMember = member;}

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            }
+
+            nbOfOtherMembers = othersActiveMembers.size();
+
+            String displayName;
+
+            if (mRoom.isInvited()) {
+                if (othersActiveMembers.size() == 1) {
+                    // Search the current user
+                    RoomMember currentUser = activeMembers.get(0);
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java]: This currentUser is useless (we got already currentUserMember)

File: matrix-sdk/src/androidTest/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        // One other user in the room
+        room = createRoom(context, withLazyLoading, 2, false);
+        Assert.assertEquals("UserName_2", room.getRoomDisplayName(context));
+
+        // 2 other users in the room
+        room = createRoom(context, withLazyLoading, 3, false);
+        Assert.assertEquals("UserName_2 and UserName_3", room.getRoomDisplayName(context));
+
+        room = createRoom(context, withLazyLoading, 4, false);
+        Assert.assertEquals("UserName_2 and 3 others", room.getRoomDisplayName(context));
```
Comment (bmarty) [inline matrix-sdk/src/androidTest/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandlerTest.java]: @Giom, this room has 4 users including me, and I am counted in the `3 others`. For a room with 3 members, I do not appear in the title. What do you think?

**Pull request merged**